providers/ebs_volume.rb

```diff
@@ -199,7 +199,11 @@ def create_volume(snapshot_id, size, availability_zone, timeout, volume_type, pi
     Timeout.timeout(timeout) do
       loop do
         vol = volume_by_id(nv[:volume_id])
-        if vol && vol[:state] != 'deleting'
+        # In some scenarios the return comes back as nil.
+        # Retry, and its all fine.
+        if vol.nil?
+          sleep 3
```

Comment:
After sleeping how will this actually retry fetching vol data again? Wont' you need to fetch the vol data once more? Otherwise you're just waiting 3 seconds and continuing on.